Doc generation fails

Generating the docs locally with `make docs`  fails with the following error:
```
± % make html                                                                                 15:42:15
\sphinx-apidoc -o ./_build/generated ../raiden/
File ./_build/generated/raiden.rst already exists, skipping.
File ./_build/generated/raiden.api.rst already exists, skipping.
File ./_build/generated/raiden.api.v1.rst already exists, skipping.
File ./_build/generated/raiden.blockchain.rst already exists, skipping.
File ./_build/generated/raiden.encoding.rst already exists, skipping.
File ./_build/generated/raiden.network.rst already exists, skipping.
File ./_build/generated/raiden.network.proxies.rst already exists, skipping.
File ./_build/generated/raiden.network.rpc.rst already exists, skipping.
File ./_build/generated/raiden.network.transport.rst already exists, skipping.
File ./_build/generated/raiden.network.transport.udp.rst already exists, skipping.
File ./_build/generated/raiden.storage.rst already exists, skipping.
File ./_build/generated/raiden.tests.rst already exists, skipping.
File ./_build/generated/raiden.tests.benchmark.rst already exists, skipping.
File ./_build/generated/raiden.tests.fixtures.rst already exists, skipping.
File ./_build/generated/raiden.tests.integration.rst already exists, skipping.
File ./_build/generated/raiden.tests.integration.api.rst already exists, skipping.
File ./_build/generated/raiden.tests.integration.contracts.rst already exists, skipping.
File ./_build/generated/raiden.tests.integration.contracts.fixtures.rst already exists, skipping.
File ./_build/generated/raiden.tests.integration.fixtures.rst already exists, skipping.
File ./_build/generated/raiden.tests.integration.long_running.rst already exists, skipping.
File ./_build/generated/raiden.tests.integration.rpc.rst already exists, skipping.
File ./_build/generated/raiden.tests.integration.transfer.rst already exists, skipping.
File ./_build/generated/raiden.tests.unit.rst already exists, skipping.
File ./_build/generated/raiden.tests.unit.api.rst already exists, skipping.
File ./_build/generated/raiden.tests.unit.transfer.rst already exists, skipping.
File ./_build/generated/raiden.tests.unit.transfer.mediated_transfer.rst already exists, skipping.
File ./_build/generated/raiden.tests.utils.rst already exists, skipping.
File ./_build/generated/raiden.transfer.rst already exists, skipping.
File ./_build/generated/raiden.transfer.mediated_transfer.rst already exists, skipping.
File ./_build/generated/raiden.ui.rst already exists, skipping.
File ./_build/generated/raiden.utils.rst already exists, skipping.
File ./_build/generated/modules.rst already exists, skipping.
sphinx-build -b html -d ./_build/doctrees   . ./_build/html
Running Sphinx v1.6.7
/Users/paul/Work/venv-raiden/lib/python3.6/site-packages/sphinx/util/compat.py:40: RemovedInSphinx17Warning: sphinx.util.compat.Directive is deprecated and will be removed in Sphinx 1.7, please use docutils' instead.
  RemovedInSphinx17Warning)
loading pickled environment... not yet created
Initiated sphinxcontrib-images backend: `sphinxcontrib_images_lightbox2.lightbox2:LightBox2`
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 45 source files that are out of date
updating environment: 45 added, 0 changed, 0 removed
docstring of raiden.api.python.RaidenAPI.set_total_channel_deposit:8: WARNING: Unexpected indentation.
docstring of raiden.api.python.RaidenAPI.set_total_channel_deposit:9: WARNING: Block quote ends without a blank line; unexpected unindent.
/Users/paul/Work/raiden/raiden/network/upnpsock.py:docstring of raiden.network.upnpsock.connect:3: WARNING: Unexpected indentation.
/Users/paul/Work/raiden/raiden/network/upnpsock.py:docstring of raiden.network.upnpsock.connect:4: WARNING: Block quote ends without a blank line; unexpected unindent.
/Users/paul/Work/raiden/raiden/network/upnpsock.py:docstring of raiden.network.upnpsock.open_port:6: WARNING: Unexpected indentation.
/Users/paul/Work/raiden/raiden/network/upnpsock.py:docstring of raiden.network.upnpsock.open_port:7: WARNING: Block quote ends without a blank line; unexpected unindent.
docstring of raiden.network.proxies.token_network.TokenNetwork.events_filter:3: WARNING: Unexpected indentation.
docstring of raiden.network.proxies.token_network.TokenNetwork.events_filter:4: WARNING: Block quote ends without a blank line; unexpected unindent.
docstring of raiden.network.rpc.client.JSONRPCClient.deploy_solidity_contract:13: WARNING: Unexpected indentation.
docstring of raiden.network.rpc.client.JSONRPCClient.deploy_solidity_contract:15: WARNING: Block quote ends without a blank line; unexpected unindent.

Exception occurred:
  File "/Users/paul/Work/venv-raiden/lib/python3.6/site-packages/sphinx/ext/napoleon/__init__.py", line 431, in _skip_member
    cls_path, _, _ = qualname.rpartition('.')
AttributeError: 'functools.partial' object has no attribute 'rpartition'
The full traceback has been saved in /var/folders/gz/0f_3f9k96gd1hblwgd1r8h280000gn/T/sphinx-err-xaiklbfi.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [html] Error 1
```

--- a/docs/conf.py
+++ b/docs/conf.py
@@ -65,12 +65,6 @@ def check_if_nightly(version):
 # ones.
 extensions = [
     'releases',
-    'sphinx.ext.autodoc',
-    'sphinx.ext.todo',
-    'sphinx.ext.coverage',
-    'sphinx.ext.mathjax',
-    'sphinx.ext.napoleon',
-    'sphinx.ext.viewcode',
     'sphinxcontrib.httpdomain',
     'sphinxcontrib.httpexample',
     'sphinxcontrib.images',

--- a/docs/rest_api.rst
+++ b/docs/rest_api.rst
@@ -74,7 +74,7 @@ Errors
 For any non-successful http status code, e.g. :http:statuscode:`409` or :http:statuscode:`400` there will be an accompanying ``errors`` field in the response json which you can check for more information on what went wrong with your request.
 However, when Raiden fails to process the incoming request and raises an exception, the returned http status code will be :http:statuscode:`500`. The caveat of this is that the response body will be just a string message which says "Internal server error".
 This is because we rely on our underlying stack to handle this while we take care of shutting down the API server preventing further incoming requests caused the exception in the first place from tampering with a state that was corrupted.
-In any way, we consider :http:statuscode:`500` errors as bugs in the Raiden client. If you encounter such errors, please report the bug `here`<https://github.com/raiden-network/raiden/issues/new?template=bug_report.md>.
+In any way, we consider :http:statuscode:`500` errors as bugs in the Raiden client. If you encounter such errors, please report the bug `here <https://github.com/raiden-network/raiden/issues/new?template=bug_report.md>`_.
 
 Endpoints
 ***********

--- a/raiden/api/python.py
+++ b/raiden/api/python.py
@@ -334,16 +334,16 @@ class RaidenAPI:
 
         Raises:
             InvalidAddress: If either token_address or partner_address is not
-            20 bytes long.
-            TransactionThrew: May happen for multiple reasons:
+                20 bytes long.
+                TransactionThrew: May happen for multiple reasons:
                 - If the token approval fails, e.g. the token may validate if
-                  account has enough balance for the allowance.
+                account has enough balance for the allowance.
                 - The deposit failed, e.g. the allowance did not set the token
-                  aside for use and the user spent it before deposit was called.
+                aside for use and the user spent it before deposit was called.
                 - The channel was closed/settled between the allowance call and
-                  the deposit call.
+                the deposit call.
             AddressWithoutCode: The channel was settled during the deposit
-            execution.
+                execution.
             DepositOverLimit: The total deposit amount is higher than the limit.
         """
         chain_state = views.state_from_raiden(self.raiden)

--- a/raiden/network/proxies/token_network.py
+++ b/raiden/network/proxies/token_network.py
@@ -958,6 +958,7 @@ class TokenNetwork:
             to_block: typing.BlockSpecification = None,
     ) -> StatelessFilter:
         """ Install a new filter for an array of topics emitted by the contract.
+
         Args:
             topics: A list of event ids to filter for. Can also be None,
                     in which case all events are queried.

--- a/raiden/network/rpc/client.py
+++ b/raiden/network/rpc/client.py
@@ -237,6 +237,7 @@ class JSONRPCClient:
     ):
         """
         Deploy a solidity contract.
+
         Args:
             sender (address): the sender address
             contract_name (str): the name of the contract to compile

--- a/raiden/network/upnpsock.py
+++ b/raiden/network/upnpsock.py
@@ -36,6 +36,7 @@ def valid_mappable_ipv4(address):
 
 def connect():
     """Try to connect to the router.
+
     Returns:
         u (miniupnc.UPnP): the connected upnp-instance
         router (string): the connection information
@@ -72,6 +73,7 @@ def connect():
 def open_port(upnp, internal_port, external_start_port=None):
     """Open a port for the raiden service (listening at `internal_port`) through
     UPnP.
+
     Args:
         internal_port (int): the target port of the raiden service
         external_start_port (int): query for an external port starting here

